common/util.go

```diff
@@ -475,3 +476,35 @@ func StructToMap(obj interface{}, output map[string]interface{}) error {
 
 	return json.Unmarshal(data, &output) // Convert to a map
 }
+
+// Concurrent Safe String keyed map
```

Comment:
Moving it to a dedicated file would be better, all the rest looks good to me. Thanks again!

Reply:
Where do you think this would fit. Kinda puzzled.

Reply:
I'm thinking same package but, in a concurrent.go file?

Reply:
Same package, `string_keyed_map.go`?

Reply:
I was thinking along the lines any common concurrent safe implementations required moving forward could be placed in concurrent.go. If this doesn't track, I will change the filename. Let me know.

Reply:
I think the word `concurrent` is too big.

Reply:
This is even bigger...

It is good to use straightforward file names.

Reply:
Oh big as in vague. `concur_safe_type.go`? Sorry if I'm dragging this naming out 😅.

Reply:
Its just that the other files in `common/` are `json.go` , `retry.go`, `string.go`, `time.go` etc are generic names

Reply:
My personal opinions:

1. The package structure in Argo Events (including file names) should be refactored;
2. Those file names (json.go, string.go ...) are kind of simple, because they provide that particular file name related utility functions (**not struct functions**), and maybe there's no better names to describe it - you can imagine we could name them something like `json_util.go` or `string_util.go`, that wouldn't make too much difference. But it's for sure it does not make much sense to separate them into different files.
3. If it's util structs (like the one you added), there is already a clear boundary (the struct itself), putting them in separated files and naming them as straightforward as possible might be better, so that ppl clearly know what they are about.

Reply:
This makes a lot of sense. Thanks for clarifying!